app/javascript/app/components/error-messages/error-messages-component.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import styles from './error-messages-styles.scss';
+
+class ErrorMessages extends PureComponent {
+  // eslint-disable-line react/prefer-stateless-function
+  render() {
+    const { errors, className } = this.props;
+    return errors && errors.length > 0 ? (
+      <div className={cx(styles.alertContainer, className)}>
+        {errors.map(errorMessage => (
+          <div className={cx(styles.alert)} key={`${errorMessage}`}>
+            {errorMessage}
+          </div>
+        ))}
+      </div>
+    ) : null;
```

Comment:
React requires to render something or null. It doesn't like rendering undefined

Reply:
it won't be rendering undefined; it will either render the component if the `hasErrors` evaluates to true or nothing. It's called a short-circuit operator - here's a thorough article about conditional rendering in React and there's a section about it https://blog.logrocket.com/conditional-rendering-in-react-c6b0e5af381e
it's the same as using a ternary operator with null as an alternative but shorter

Reply:
The short-circuit operator will work inside some render object in react. The react components expect to return either a node or null. Using it here will result in the following error:

```Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.```

Reply:
Thanks @aabdaab for the reading! I didn't realised this:
`One advantage of returning null instead of an empty element is that you’ll improve a little bit the performance of your app because React won’t have to unmount the component to replace it. `

Reply:
ok, it was driving me crazy and I finally know what's happening. you're right the `hasErrors` can evaluate to undefined if the `ErrrorMessages` errors prop is getting errors as undefined. One way to solve this would be to define `defaultProps` for error messages in the components that pass errors so in our case `input-component` and `sign-up-component`. I'd highly vote for that as also eslint require-default-props is recommending defining default props for every prop that is not marked as required --> https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md